### PR TITLE
Allow initial View in Calendar Card

### LINKED ISSUE #7038: Calendar card: Allow choosing default view
## The request

Add an option to set the default view for the Calendar card. Either Monthly, Weekly, or Daily. Current default is Monthly.

**Background:**
The default monthly view is the least valuable to me on my dashboard (mostly due to massive space required to properly see events). The weekly view would be much preferred but each time the dashboard re-loads I have to re-select it manually which reduces a lot of the value of having it on my wall panel. The family can't take a quick glance at the board and see what's coming up today or for the week.

## The alternatives

Community add-ons which I'd rather not get into.

### PR DESCRIPTION
**Pull request by zsarnett**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
inital_view: dayGridMonth | listWeek | dayGridDay
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request: TODO

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/calendar/ha-full-calendar.ts [hunk 1]
```diff
@@ -189,6 +194,9 @@ class HAFullCalendar extends LitElement {
       locale: this.hass.language,
     };
 
+    config.initialView = this.initialView;
```
Comment (bramkragten) [inline src/panels/calendar/ha-full-calendar.ts:197]:
```suggestion
      locale: this.hass.language,
      initialView: this.initialView
    };
```

Comment (bramkragten) [inline src/panels/calendar/ha-full-calendar.ts:197]: Or
```suggestion
      locale: this.hass.language,
    };
	if (this.initalView) {
    	config.initialView = this.initialView;
    }
```

File: src/panels/lovelace/cards/hui-calendar-card.ts [hunk 2]
```diff
@@ -91,7 +89,7 @@ export class HuiCalendarCard extends LitElement implements LovelaceCard {
       };
     });
 
-    this._config = config;
+    this._config = { initial_view: "dayGridMonth", ...config };
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-calendar-card.ts:92]: The default already is `dayGridMonth`, so this is not needed?

**Comment by cadavre** [comment]
@zsarnett really looking forward to get this merged!

**Comment by bramkragten** [comment]
> @zsarnett really looking forward to get this merged!

Please just add a thumbs-up reaction for things like this. Now everyone is pinged.

File: src/panels/lovelace/cards/hui-calendar-card.ts (hunk 2 above)
Comment (zsarnett) [reply src/panels/lovelace/cards/hui-calendar-card.ts:92]: Well, if `this._config.initialView` is `undefined` then we set it on the calendar as `undefined`. The property default gets overridden by `Undefined`. `ifDefined` directive is only for attributes I believe (or I didn't set it up correctly) so the default here is needed unless there is another way I have missed (which I am sure there is)

**Review by bramkragten — APPROVED** [review]

**Pull request merged**